src/webgpu/api/validation/buffer_mapping.spec.ts

```diff
@@ -0,0 +1,763 @@
+          offset: 0,
+          size: 80 + kSizeAlignement,
+        },
+        {
+          bufferSize: 80,
+          mapOffset: undefined,
+          mapSize: undefined,
+          offset: kOffsetAlignment,
+          size: 80,
+        },
+
+        // Test for a mapAsync call with an implicit size
+        { bufferSize: 80, mapOffset: 24, mapSize: undefined, offset: 24, size: 80 - 24 },
+        {
+          bufferSize: 80,
+          mapOffset: 24,
+          mapSize: undefined,
+          offset: 0,
+          size: 80 - 24 + kSizeAlignement,
+        },
+        {
+          bufferSize: 80,
+          mapOffset: 24,
+          mapSize: undefined,
+          offset: kOffsetAlignment,
+          size: 80 - 24,
+        },
+
+        // Test for a non-empty buffer fully mapped.
+        { bufferSize: 80, mapOffset: 0, mapSize: 80, offset: 0, size: 80 },
+        { bufferSize: 80, mapOffset: 0, mapSize: 80, offset: kOffsetAlignment, size: 80 },
+        { bufferSize: 80, mapOffset: 0, mapSize: 80, offset: 0, size: 80 + kSizeAlignement },
+
+        { bufferSize: 80, mapOffset: 0, mapSize: 80, offset: 40, size: 40 },
+        { bufferSize: 80, mapOffset: 0, mapSize: 80, offset: 40 + kOffsetAlignment, size: 40 },
+        { bufferSize: 80, mapOffset: 0, mapSize: 80, offset: 40, size: 40 + kSizeAlignement },
+
+        // Test for a buffer partially mapped.
+        { bufferSize: 80, mapOffset: 24, mapSize: 40, offset: 24, size: 40 },
+        { bufferSize: 80, mapOffset: 24, mapSize: 40, offset: 24 - kOffsetAlignment, size: 40 },
+        { bufferSize: 80, mapOffset: 24, mapSize: 40, offset: 24 + kOffsetAlignment, size: 40 },
+        { bufferSize: 80, mapOffset: 24, mapSize: 40, offset: 24, size: 40 + kSizeAlignement },
+
+        // Test for a partially mapped buffer with implicit size and offset for getMappedRange.
+        // - Buffer partially mapped in the middle
+        { bufferSize: 80, mapOffset: 24, mapSize: 40, offset: undefined, size: undefined },
+        { bufferSize: 80, mapOffset: 24, mapSize: 40, offset: 24, size: undefined },
+        // - Buffer partially mapped to the end
+        { bufferSize: 80, mapOffset: 24, mapSize: undefined, offset: 24, size: undefined },
+        // - Buffer partially mapped from the start
+        { bufferSize: 80, mapOffset: 0, mapSize: 64, offset: undefined, size: undefined },
+        { bufferSize: 80, mapOffset: 0, mapSize: 64, offset: undefined, size: 64 },
+      ])
+  )
+  .fn(async t => {
+    const { mapMode, bufferSize, mapOffset, mapSize, offset, size } = t.params;
+    const buffer = t.createMappableBuffer(mapMode, bufferSize);
+    await buffer.mapAsync(mapMode, mapOffset, mapSize);
+
+    const actualMapOffset = mapOffset ? mapOffset : 0;
+    const actualMapSize = mapSize ? mapSize : bufferSize - actualMapOffset;
+
+    const actualOffset = offset ? offset : 0;
+
+    let success = actualOffset >= actualMapOffset && actualOffset <= bufferSize;
+    if (size !== undefined) {
```

Comment:
Fixed the condition. The tests weren't failing when success was true but `getMappedRange` failed because the test function was async and the exception was thrown in the returned promise.